src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -311,26 +338,29 @@ async Task archiveTable(string name, string context)
                     throw new CreateContainerException(message);
                 }
             }
-
-            var container = new Container(id, StreamMode);
-            var tasks = new List<Task> { archiveTable(container.PrimaryTable, "Data") };
-            foreach (var table in container.Tables)
-                if (await CheckTableExists(executionContext, providerDefinitionId, table.Value.Name))
-                    tasks.Add(archiveTable(table.Value.Name, table.Key));
-
-            await Task.WhenAll(tasks);
         }
 
         public override async Task RenameContainer(ExecutionContext executionContext, Guid providerDefinitionId,
             string id, string newName)
         {
             var config = await base.GetAuthenticationDetails(executionContext, providerDefinitionId);
+            var schema = config.GetSchema();
+            var currentContainer = new Container(id, StreamMode);
+            var updatedContainer = new Container(newName, StreamMode);
 
-            async Task renameTable(string currentName, string updatedName, string context)
+            var tasks = new List<Task>
             {
-                var tempName = SqlStringSanitizer.Sanitize(updatedName);
+                renameTable(currentContainer.PrimaryTable, updatedContainer.PrimaryTable, "Data")
+            };
+            foreach (var current in currentContainer.Tables)
+                if (await CheckTableExists(executionContext, providerDefinitionId, current.Value.Name.GetValue()))
+                    tasks.Add(renameTable(current.Value.Name, updatedContainer.Tables[current.Key].Name, current.Key));
 
-                var sql = BuildRenameContainerSql(currentName, tempName, out var param);
+            await Task.WhenAll(tasks);
+
+            async Task renameTable(SanitizedSqlName currentName, SanitizedSqlName updatedName, string context)
```

Comment:
Agree, same as above
"agree about separate internal methods or even just a straightforward method body instead of local functions. Unfortunately haven't had time to properly refactor this.
At least not in scope of this task."